projects/common/src/time/time-duration.ts

```diff
@@ -14,16 +14,55 @@ export class TimeDuration {
     return this.millis;
   }
 
-  public static parse(durationString: string): TimeDuration {
-    return new TimeDuration(toSeconds(parse(durationString)), TimeUnit.Second);
+  /**
+   * Method to convert a duration string to TimeDuration object
+   * Currently we store the information as a single unit string.
+   * Will need to be updated to handle multiple units when needed.
+   */
+  public static parse(durationString: string): TimeDuration | undefined {
+    const duration: Duration = parse(durationString);
+    const isValid = (durationValue: number | undefined) => durationValue !== undefined && durationValue !== 0;
+    if (isValid(duration.years)) {
+      return new TimeDuration(duration.years!, TimeUnit.Year);
```

Comment:
Since this is reading only the most significant portion of the duration, my concern is that this will make it really easy to introduce a bug - for example, imagine the server returns a duration of PT2H30M. So, rather than changing our duration to hold multiple units (which I suspect is out of scope right now), could we disregard trying to preserve the exact units of the duration string? If we're ok storing it in millis, we can change this logic to be something like
```typescript
   const totalMillis = new TimeDuration(duration.years ?? 0, TimeUnit.Year).toMillis()
   + new TimeDuration(duration.months ?? 0, TimeUnit.Month).toMillis()
   + ...

return new TimeDuration(totalMillis, TimeUnit.Millisecond);
```
The formatted strings can still handle converting that back to its most significant units.